homeassistant/components/tuya/fan.py

```diff
@@ -71,27 +71,35 @@ def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> Non
                 self.device.function[DPCode.MODE].values
             ).get("range", [])
 
+        self.fan_speed_range_len = 0
+        self.fan_speed_range_enum = []
+
         # Air purifier fan can be controlled either via the ranged values or via the enum.
         # We will always prefer the enumeration if available
         #   Enum is used for e.g. MEES SmartHIMOX-H06
         #   Range is used for e.g. Concept CA3000
-        self.air_purifier_speed_range_len = 0
-        self.air_purifier_speed_range_enum = []
-        if self.device.category == "kj" and (
+        #
+        # for FSD based fans, Enums are also used, under DPCode.FAN_SPEED
+
+        if (self.device.category == "fsd" or self.device.category == "kj") and (
```

Comment:
We should leverage entity descriptions instead of adding if/else constructs here.

Reply:
This should now be addressed.

Reply:
It isn't as your latest commit doesn't do anything with entity descriptions.